fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -188,6 +282,42 @@ public void gsonPostProcess() throws IOException {
             // it is checked when creation
             ((OlapTable) table).addRelatedMaterializedView(id);
         }
+        // analyze expression, because it converts to sql for serialize
+        ConnectContext connectContext = new ConnectContext();
+        connectContext.setCluster(SystemInfoService.DEFAULT_CLUSTER);
+        connectContext.setDatabase(db.getFullName());
+        // set privilege
+        connectContext.setQualifiedUser(Auth.ROOT_USER);
+        connectContext.setCurrentUserIdentity(UserIdentity.ROOT);
+        PartitionInfo partitionInfo = this.getPartitionInfo();
+        if (partitionInfo instanceof SinglePartitionInfo) {
+            return;
+        }
+        ExpressionRangePartitionInfo expressionRangePartitionInfo = (ExpressionRangePartitionInfo) partitionInfo;
+        // currently, mv only supports one expression
+        Expr partitionExpr = expressionRangePartitionInfo.getPartitionExprs().get(0);
+        try {
+            QueryStatement queryStatement = ((QueryStatement) SqlParser.parse(
+                    this.viewDefineSql, connectContext.getSessionVariable().getSqlMode()).get(0));
+            Analyzer.analyze(queryStatement, connectContext);
+            Map<String, TableRelation> tableRelations = AnalyzerUtils.collectAllTableRelation(queryStatement);
+            List<Field> fields = Lists.newArrayList();
+            for (TableRelation value : tableRelations.values()) {
+                fields.addAll(value.getRelationFields().getAllFields());
+            }
+            Scope scope = new Scope(RelationId.anonymous(), new RelationFields(fields));
+            if (partitionExpr instanceof FunctionCallExpr) {
+                FunctionCallExpr functionCallExpr = (FunctionCallExpr) partitionExpr;
+                if (functionCallExpr.getFn() == null) {
+                    // set fn into functionCallExpr, because FunctionCallExpr serialized use sql
+                    ExpressionAnalyzer.analyzeExpression(functionCallExpr, new AnalyzeState(),
+                            scope, connectContext);
+                }
+            }
+        } catch (ParsingException parsingException) {
```

Comment:
should catch SemanticException, if the sql parsed successfully once, shouldn't failed again, but if the table was dropped, column was changed, Analyzing will maybe fail.